frontend/src/pages/AccountReceive.vue

```diff
@@ -37,14 +37,14 @@
               <div class="row justify-start q-col-gutter-md">
```

Comment:
The error was fixed in https://github.com/ScopeLift/umbra-protocol/pull/549/commits/194834e49f178ff58dcb219493fc201df6e508bc Seems like in prod we scan for all announcements when scanning for a not yet registered address. Are we sure we want to just scan for an empty list of announcements? e.g. I'm thinking of the rare cases when a user registers stealth keys twice and how that might change the receive table.

Reply:
I was just thinking of the edge case we received on Discord -- where the user accidentally re-registers stealth keys and when they scan, they only see their previous txs but not the latest tx that they've sent after they're re-registered. I was wondering if returning an empty announcement[] will cause even their previous txs to disappear from the receive table. I hope that made sense, but I think it's too much of an edge case anyways and agree that it would be a huge waste for most unregistered users to scan for all announcements.

Made the changes here: https://github.com/ScopeLift/umbra-protocol/pull/549/commits/62605e02ed9a75468eda928ed700de7e3c5e85e5

Reply:
> where the user accidentally re-registers stealth keys and when they scan, they only see their previous txs but not the latest tx that they've sent after they're re-registered.

The reason the users saw different things in this case is because they are scanning with a different key. For a user tha has called setStealthKeys more than one time, the correct behavior for this PR would be to fetch and scan all announcements from the first time they called it. Is this what the PR is doing?

> I was wondering if returning an empty announcement[] will cause even their previous txs to disappear from the receive table.

Well, it would. But why would we return an empty array for these users. The only time we should return an empty list is if the user has *never* registered stealth keys. Is that what we're doing?

Reply:
> The reason the users saw different things in this case is because they are scanning with a different key. For a user tha has called setStealthKeys more than one time, the correct behavior for this PR would be to fetch and scan all announcements from the first time they called it. Is this what the PR is doing?

Yes, https://github.com/ScopeLift/umbra-protocol/pull/549#discussion_r1251144933 we're using index 0 of the `stealthKeyLogs` which is first time someone registering a stealth key.

> Well, it would. But why would we return an empty array for these users. The only time we should return an empty list is if the user has _never_ registered stealth keys. Is that what we're doing?

Yes, if `stealthKeyLogs` see the user hasn't registered and if they haven't specified a start block in advanced mode, it will return an empty array.

> I was wondering if returning an empty announcement[] will cause even their previous txs to disappear from the receive table.

Therefore, I think I was wrong to assume this, and it's a non-issue.

Reply:
> If the user has *never* registered a stealth key, and they're not scanning with a private key in advance mode, then there's no chance they've ever received a payment and there's no point in doing the scanning.

Technically there is still a chance they received a payment. They could have signed the same message and generated the keys (with the umbra SDK, or by just copying out the logged pubkeys from the console on the receive page), and just not registered those keys on-chain. Then they share those keys with the sender directly, which the sender pastes into the app or again used the SDK. Though, this is very unlikely and may have never happened, but just want to clarify it's not "no chance" (unless I misunderstood something)

Reply:
Yes this is technically correct @mds1, but at that point we have to assume they are experts who know what they're doing. They could still force the scan with advanced mode -> custom block number.